impeller/renderer/render_target_builder.h

```diff
@@ -0,0 +1,100 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <optional>
+#include "context.h"
+#include "flutter/fml/macros.h"
+#include "impeller/geometry/size.h"
+#include "impeller/renderer/allocator.h"
+#include "impeller/renderer/formats.h"
+#include "render_target.h"
+
+namespace impeller {
+
+enum class RenderTargetType { Offscreen, OffscreenMSAA, kUnknown };
+
+class RenderTargetBuilder {
+ private:
+  ISize size_;
+  std::string label_;
+
+  StorageMode color_storage_mode_;
```

Comment:
Default all ivars in the header with sensible defaults. Perhaps a device-transient texture.